src/components/TodoListScreen.js

```diff
@@ -11,31 +11,38 @@ import {
 } from 'react-native';
 
 import { Icon } from 'react-native-elements';
-import {removeTodo} from '../store/TodosActions';
+import {removeTodo, showTodos} from '../store/TodosActions';
 
 
 class TodoListScreen extends Component {
-  onRemoveTodo = (index) => {
-    const { removeTodo } = this.props;
+  componentDidMount(){
+    const { showTodos } = this.props;
+    showTodos();
+  }
+
+  onRemoveTodo = (id) => {
+    const { removeTodo, showTodos } = this.props;
 
-    removeTodo(index);
+    removeTodo(id);
+    showTodos();
   } 
 
   render(){
     const { navigation, allTodos} = this.props;
+    console.log(allTodos);
 
     return (
       <ScrollView contentContainerStyle={styles.container} >
             <Text style={{ fontSize: 30, height: "10%" }}>To Do</Text>
-            {allTodos.map((todo, index) => (
-              <View key={index} style = {{flexDirection: 'row'}}>
-                <Text style={styles.todoItem}>{todo}</Text>
+            {allTodos.map(todo => (
```

Comment:
Aqui podes usar destruct no **todo**
```suggestion
            {allTodos.map(({ id, text }) => (
```